USAGE.md

```diff
@@ -0,0 +1,274 @@
+    + [Intermediate-Only](#intermediate-only)
+
+## Terminology
+
+| Term | Definition |
+| ---- | ---------- |
+| ECS | Elastic Common Schema. For the purposes of this guide, ECS may refer to either the schema itself or the repo/tooling used to maintain the schema |
+| artifacts | Various kinds of files or programs that can be generated based on ECS |
+| field set | Groups of related fields in ECS |
+| schema | Another term for a group of related fields in ECS. Used interchangeably with field set |
+| schema definition | The markup used to define a schema in ECS |
+| attributes | The properties of a field or field set that are used to define that field or field set in a schema definition |
+
+## Setup and Install
+
+### Prerequisites
+
+* [Python 3.6+](https://www.python.org/)
+* [make](https://www.gnu.org/software/make/)
+* [pip](https://pypi.org/project/pip/)
+* [git](https://git-scm.com/)
+
+#### Clone from GitHub
+
+The recommended way to download the ECS repo is `git clone`:
+
+```
+$ git clone https://github.com/elastic/ecs
+$ cd ecs
+```
+
+Prior to installing dependencies or running the tools, it's recommended to check out the `git` branch for the ECS version being targeted.
+
+**Example**: For ECS `1.5.0`:
+
+```
+$ git checkout v1.5.0
+```
+
+#### Option 1: Install dependencies via make (recommended)
+
+Setting up a `virtualenv` (`venv`) can be accomplished by running `make ve` the top-level of the ECS repo:
+
+```
+$ make ve
+```
+
+All necessary Python dependencies will also be installed with `pip`.
+
+#### Option 2: Install dependencies via pip
+
+Install dependencies using `pip` (An active `virutalenv` is recommended):
+
+```
+$ pip install -r scripts/requirements.txt
+```
+
+## Usage
+
+### Getting Started - Generating Artifacts
+
+Using the defaults, the [generator](scripts/generator.py) script generates the artifacts based on the [current](schemas) ECS schema.
+
+```
+$ python scripts/generator.py
+Loading schemas from local files
+Running generator. ECS version 1.5.0
+```
+
+**Points to note on the defaults**:
+
+* Artifacts are created in the [`generated`](generated) directory and the entire schema is included
+* Documentation updates will be written to the appropriate file under the `docs` directory. More specifics on generated doc files is covered in the [contributor's file](https://github.com/elastic/ecs/blob/master/CONTRIBUTING.md#generated-documentation-files)
+* Each run of the script will rewrite the entirety of the `generated` directory
+* The script will need to be executed from the top-level of the ECS repo
+* The `version` displayed when running `generator.py` is based on the current value of the [version](version) file in the top-level of the repo
+
+The following options add functionality beyond the defaults.
+
+### Generator Options
+
+#### Out
+
+Generate the ECS artifacts in a different output directory. The argument expects to be passed an existing directory:
```

Comment:
The generator script will actually create the path if it doesn't exist: https://github.com/elastic/ecs/blob/master/scripts/generator.py#L44

Reply:
Thanks for catching! Fixed.